edit_questionpy_form.php

```diff
@@ -82,6 +82,11 @@ protected function definition_package_settings(MoodleQuickForm $mform, string $p
         $packagearray = $package->as_localized_array($languages);
         $packagearray['selected'] = true;
         $packagearray['versions'] = ['hash' => $pkgversion->hash, 'version' => $pkgversion->version];
+        $packagearray['contextid'] = $PAGE->context->id;
+
+        $usercontext = context_user::instance($USER->id);
+        $ufservice = \core_favourites\service_factory::get_service_for_user_context($usercontext);
+        $packagearray['isfavourite'] = $ufservice->favourite_exists('qtype_questionpy', 'package', $package->id, $usercontext);
```

Comment:
Vielleicht wäre es sinnvoll, den Code zum Abrufen und Ändern des favourite-Status in die Package Klasse zu verschieben. Oder in eine eigene Helper-Klasse. Du benutzt den ja sowohl hier als auch in dem Webservice.